src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -11645,61 +11640,49 @@ public String incrementHealth(float increment) {
 	}
 
 	public String incrementHealth(GameCharacter attacker, float increment) {
-		// Fetishes:
 		if(Main.game.isInCombat()) {
-			// Masochist:
-			if (isMasochist() && increment < 0) {
-				
-				this.setHealth(getHealth() + (increment*0.75f));
-				
-				if(increment<0) {
-					Combat.incrementTotalDamageTaken(this, -increment*0.75f);
+			if (increment < 0) {
+				//Target of Attack is Masochistic
+				if (isMasochist()) {
+					increment *= 0.75f;
+					this.incrementFetishExperience(Fetish.FETISH_MASOCHIST, 2);	
 				}
-				
-				this.incrementFetishExperience(Fetish.FETISH_MASOCHIST, 2);
-
-				float manaLoss = (Math.round((-increment*0.25f)*10))/10f;
-				manaLoss = Attack.getModifiedDamage(null, this, Attack.SEDUCTION, DamageType.LUST, manaLoss);
-				
-				return (UtilText.parse(this,
-						"<p>"
-							+ "Due to [npc.namePos] <b style='color:" + Colour.GENERIC_ARCANE.toWebHexString() + ";'>masochist fetish</b>, incoming damage is reduced by 25%, but in turn, [npc.she] [npc.verb(take)]"
-							+ " <b>"+(manaLoss)+"</b> <b style='color:" + Attribute.DAMAGE_LUST.getColour().toWebHexString() + ";'>lust damage</b> as [npc.she] [npc.verb(struggle)] to control [npc.her] arousal!"
-						+ "</p>"))
-						+incrementLust(manaLoss, false);
-				
-			// Sadist:
-			} else if (attacker!=null && attacker.hasFetish(Fetish.FETISH_SADIST) && increment < 0) {
-				float manaLoss = (Math.round((-increment*0.1f)*10))/10f;
-				manaLoss = Attack.getModifiedDamage(null, attacker, Attack.SEDUCTION, DamageType.LUST, manaLoss);
-				
-				attacker.incrementFetishExperience(Fetish.FETISH_SADIST, 2);
-
-				this.setHealth(getHealth() + (increment*1.05f));
-
-				if(increment<0) {
-					Combat.incrementTotalDamageTaken(this, -increment*1.05f);
+				//Attacker is Sadistic
+				if (attacker.hasFetish(Fetish.FETISH_SADIST) && attacker != null) {
```

Comment:
This should be the other way around, first check if the attacker exists and then if it has the fetish.

Reply:
this does not matter to neither runtime nor Compiler.
both are checked separately and then logically AND added.

Reply:
It does matter because the `&&` operator is left-associative and short-circuiting.
This means that the left-most condition is evaluated first and subsequent conditions are only evaluated if the previous one was `true`.
In this case it means that the code checks if `attacker.hasFetish(Fetish.FETISH_SADIST)` is true AND THEN checks if `attacker != null` or in other words, first it checks if the attacker has the fetish and then if it exists.

You can easily make a test case for this if you need further evidence.